lib/active_model_serializers/adapter/json_api/pagination_links.rb

```diff
@@ -2,14 +2,21 @@ module ActiveModelSerializers
   module Adapter
     class JsonApi < Base
       class PaginationLinks
+        MissingSerializationContextError = Class.new(KeyError)
         FIRST_PAGE = 1
 
         attr_reader :collection, :context
 
         def initialize(collection, adapter_options)
           @collection = collection
           @adapter_options = adapter_options
-          @context = adapter_options.fetch(:serialization_context)
+          @context = adapter_options.fetch(:serialization_context) do
+            fail MissingSerializationContextError, <<-EOF.freeze
+ JsonApi::PaginationLinks requires a ActiveModelSerializers::SerializationContext.
+ Please pass a ':serialization_context' option or
+ override CollectionSerializer#paginated? to return 'false'.
```

Comment:
Well, it's only enabled for paginatable objects, so saying 'def paginated? false; end doesn't seem terrible to me.  Maybe a serializer option 'paginate: false'?  I wonder if that's already a thing in other gems...

B mobile phone

> On Jun 9, 2016, at 4:05 AM, Lucas Hosseini <notifications@github.com> wrote:
> 
> In lib/active_model_serializers/adapter/json_api/pagination_links.rb:
> 
> >          FIRST_PAGE = 1
> >  
> >          attr_reader :collection, :context
> >  
> >          def initialize(collection, adapter_options)
> >            @collection = collection
> >            @adapter_options = adapter_options
> > -          @context = adapter_options.fetch(:serialization_context)
> > +          @context = adapter_options.fetch(:serialization_context) do
> > +            fail MissingSerializationContextError, <<-EOF.freeze
> > + JsonApi::PaginationLinks requires a ActiveModelSerializers::SerializationContext.
> > + Please pass a ':serialization_context' option or
> > + override CollectionSerializer#paginated? to return 'false'.
> That sucks for the poor guy who just does not care about pagination. Shouldn't we disable pagination by default?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>